### PR TITLE
Create settings page to change rover settings

### PR DESCRIPTION
**Pull request by cabarnes**
More of #19. This allows for changing the rover pin settings. There are a couple disabled fields for features that will be added later.
image

### CODE REVIEW

File: mission_control/templates/rover_settings.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      <select class="form-control" id="sbc" disabled>
+        <option>Next Thing Co. CHIP</option>
+        <option>Raspberry Pi</option>
+        <option>Beaglebone Black</option>
+      </select>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-md-12">
+      <h4>{% trans "Pin assignments:" %}</h4>
```
Comment (aninternetof) [inline mission_control/templates/rover_settings.html]: Maybe just "Assignments"? The `name` field is in this section, and it's not really a pin assignment.

File: mission_control/templates/rover_settings.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-md-12">
+      <form method="post">
+        {% csrf_token %}
+        <div class="form-group row">
+          {{ form.as_p }}
```
Comment (aninternetof) [inline mission_control/templates/rover_settings.html]: Had to look up `as_p`. That's cool that they give you the option.

File: mission_control/tests/test_urls.py [hunk 3]
```diff
@@ -76,3 +76,15 @@ def test_block_diagrams_resolve(self):
         self.assertEqual(
             resolve('/mission-control/block-diagrams/').view_name,
             'mission-control:blockdiagram-list')
+
+    def test_rover_settings_reverse(self):
+        """'mission-control:rover_settings' should reverse to /mission-control/rover-settings/.""" #noqa
```
Comment (aninternetof) [inline mission_control/tests/test_urls.py]: Is the `#noqa` for the line-length?

File: mission_control/views.py [hunk 4]
```diff
@@ -36,6 +38,28 @@ def rover_list(request):
...
+    rover = get_object_or_404(Rover, owner=request.user, pk=pk)
+    print(request.POST)
+    if request.method == 'POST':
+        form = RoverForm(instance=rover, data=request.POST)
+
+        if form.is_valid():
+            form.save()
+            return redirect(reverse('mission-control:rover_list'))
+
+        form = RoverForm(instance=rover)
```
Comment (aninternetof) [inline mission_control/views.py]: If the form is not valid, then we return the user to a new, blank form?
That's good with me -- just making sure I follow.

File: mission_control/templates/rover_settings.html (hunk 1 above)
Comment (cabarnes) [inline mission_control/templates/rover_settings.html]: Good point. I'll make the change.

File: mission_control/tests/test_urls.py (hunk 3 above)
Comment (cabarnes) [inline mission_control/tests/test_urls.py]: Correct. The linter requires it to be a single line but it's too long. I think being this specific is a good idea though so it's a compromise.

File: mission_control/views.py (hunk 4 above)
Comment (cabarnes) [inline mission_control/views.py]: Yeah. The form is created by Django so it really should always be valid. But, if somehow it's not, it'll stay on the page and populate the fields from the saved settings.

**Pull request merged**